### PR TITLE
support docker-compose.yaml, fix #44

### LINKED ISSUE #44: Support for docker-compose.yaml
```bash
[vagrant@compose ~]$ ls
docker-compose.yaml
[vagrant@compose ~]$ 
[vagrant@compose ~]$ henge kubernetes
file "docker-compose.yml" not found
```

The current directory is currently being searched for a `docker-compose.yml` and not `docker-compose.yaml`, how about adding support for that?

### PR DESCRIPTION
**Pull request by containscafeine**
@surajssd @kadel, updated the PR.

### CODE REVIEW

File: henge.go [hunk 1]
```diff
@@ -17,8 +17,17 @@ func main() {
 		os.Exit(1)
 	}
 
-	// check if files exists
-	err = utils.CheckIfFileExists(vals.Files)
+	// check if files exists, if multiple files specified, use the first
+	// one and break out of the loop
```
Comment (surajssd) [inline henge.go]: But we do accept multiple docker-compose files, this will hinder that?

Comment (kadel) [inline henge.go]: I have same concert as @surajssd.

Comment (containscafeine) [inline henge.go]: Will fix this once #43 is fixed!

**Comment by surajssd** [comment]
In libcompose `docker-compose.yml` is default https://github.com/docker/libcompose/blob/master/cli/command/command.go#L341

File: pkg/cmd/root.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ func Execute() (*types.CmdValues, error) {
 
 func addProviderFlags(cmd *cobra.Command, vals *types.CmdValues) *cobra.Command {
 
-	cmd.Flags().StringSliceVarP(&vals.Files, "files", "f", []string{"docker-compose.yml"}, "Provide docker-compose files, comma separated.")
+	cmd.Flags().StringSliceVarP(&vals.Files, "files", "f", nil, "Provide docker-compose files, comma separated.")
```
Comment (surajssd) [inline pkg/cmd/root.go]: putting `docker-compose.yml` here will be helpful to user, because it will show up in automatic help.

Comment (containscafeine) [inline pkg/cmd/root.go]: But it does not show `docker-compose.yaml`, so I removed it from here.

I'd say we add to the usage field that currently says `Provide docker-compose files, comma separated` to say, `Provide docker-compose files, comma separated, takes docker-compose.yml or docker-compose.yaml if no file is specified`

File: pkg/cmd/root.go [hunk 3]
```diff
@@ -71,3 +71,10 @@ func errorIfFileDoesNotExist(val *types.CmdValues) {
 		os.Exit(1)
 	}
 }
+
+func setDefaultComposeFile(val *types.CmdValues) {
+	val.Files = []string{"docker-compose.yml"}
+	if utils.CheckIfFileExists(val.Files) != nil {
+		val.Files = []string{"docker-compose.yaml"}
```
Comment (surajssd) [inline pkg/cmd/root.go]: This will make `docker-compose.yaml` and when this file is not found it will throw error for `docker-compose.yaml`.

Comment (surajssd) [inline pkg/cmd/root.go]: Or do one thing like also check if `docker-compose.yaml` exists and if it does not give a generic error like docker-compose files not found, that would remove this from being showing docker-compose.yaml does not exists.

File: pkg/cmd/root.go (hunk 2 above)
Comment (surajssd) [inline pkg/cmd/root.go]: Yes it makes sense to update the help and keeping it nil.

File: pkg/cmd/root.go [hunk 4]
```diff
@@ -57,17 +56,32 @@ func Execute() (*types.CmdValues, error) {
 
 func addProviderFlags(cmd *cobra.Command, vals *types.CmdValues) *cobra.Command {
 
-	cmd.Flags().StringSliceVarP(&vals.Files, "files", "f", []string{"docker-compose.yml"}, "Provide docker-compose files, comma separated.")
+	cmd.Flags().StringSliceVarP(&vals.Files, "files", "f", nil, "Provide docker-compose files, comma separated. Takes docker-compose.yml or docker-compose.yaml if no file is specified.")
```
Comment (surajssd) [inline pkg/cmd/root.go]: The usage message better can be made 

`Provide comma separated docker-compose files. Default: 'docker-compose.yml' or 'docker-compose.yaml'`

Comment (containscafeine) [inline pkg/cmd/root.go]: Done!

**Comment by surajssd** [comment]
LGTM

File: pkg/cmd/openshift.go [hunk 5]
```diff
@@ -30,6 +34,7 @@ func openshiftCmd(vals *types.CmdValues) *cobra.Command {
 				fmt.Fprintln(os.Stderr, err.Error())
 				os.Exit(1)
```
Comment (kadel) [inline pkg/cmd/openshift.go]: can we just return err? instead of printing and exiting? same thing as on line 22

File: pkg/cmd/root.go [hunk 6]
```diff
@@ -57,17 +56,32 @@ func Execute() (*types.CmdValues, error) {
...
+	if val.Files == nil {
+		val.Files = []string{"docker-compose.yml"}
+		err1 := utils.CheckIfFileExists(val.Files)
+
+		if err1 == nil {
+			return nil
+		}
+
+		val.Files = []string{"docker-compose.yaml"}
+		err2 := utils.CheckIfFileExists(val.Files)
```
Comment (rtnpro) [inline pkg/cmd/root.go]: ``err1``, ``err2`` does not look great :)

File: pkg/cmd/openshift.go (hunk 5 above)
Comment (containscafeine) [inline pkg/cmd/openshift.go]: @kadel, upon returning err, cobra prints out the help alongside the error too, which will make the output more convoluted. @surajssd and me had this discussion today morning.
How do you suggest we go about this?

File: pkg/cmd/root.go (hunk 6 above)
Comment (containscafeine) [inline pkg/cmd/root.go]: Whoops, that crept in from a previous iteration. Thanks for pointing that out.

File: pkg/cmd/openshift.go (hunk 5 above)
Comment (kadel) [inline pkg/cmd/openshift.go]: You you are right, with help it is hard to see error message :(
Keep it that way.
Can you maybe just add comment that this is reason for os.Exit there?

Comment (containscafeine) [inline pkg/cmd/openshift.go]: Done

**Comment by containscafeine** [comment]
Added a couple of simple tests for this, added the comment. Will squash once you guys review, and LGTM!

**Comment by kadel** [comment]
:+1:  LGTM

**Comment by kadel** [comment]
great work @containscafeine

**Pull request merged**